usaspending_api/download/management/commands/delta_downloads/builders.py

```diff
@@ -383,9 +385,89 @@ def account_balances(self) -> DataFrame:
+            "USSGL480200_undelivered_orders_obligations_prepaid_advanced_FYB": lambda col: sf.sum(col).alias(col),
+            "USSGL488200_upward_adj_prior_year_undeliv_orders_oblig_prepaid": lambda col: sf.sum(col).alias(col),
+            "gross_outlays_delivered_orders_paid_total": lambda col: sf.sum(col).alias(col),
+            "gross_outlays_delivered_orders_paid_total_FYB": lambda col: sf.sum(col).alias(col),
+            "USSGL490200_delivered_orders_obligations_paid": lambda col: sf.sum(col).alias(col),
+            "USSGL490800_authority_outlayed_not_yet_disbursed": lambda col: sf.sum(col).alias(col),
+            "USSGL490800_authority_outlayed_not_yet_disbursed_FYB": lambda col: sf.sum(col).alias(col),
+            "USSGL498200_upward_adj_of_prior_year_deliv_orders_oblig_paid": lambda col: sf.sum(col).alias(col),
+            "deobligations_or_recoveries_or_refunds_from_prior_year": lambda col: sf.sum(col).alias(col),
+            "USSGL487100_downward_adj_prior_year_unpaid_undeliv_orders_oblig": lambda col: sf.sum(col).alias(col),
+            "USSGL497100_downward_adj_prior_year_unpaid_deliv_orders_oblig": lambda col: sf.sum(col).alias(col),
+            "USSGL487200_downward_adj_prior_year_prepaid_undeliv_order_oblig": self.filter_and_sum,
+            "USSGL497200_downward_adj_of_prior_year_paid_deliv_orders_oblig": self.filter_and_sum,
+            "USSGL483100_undelivered_orders_obligations_transferred_unpaid": lambda col: sf.sum(col).alias(col),
+            "USSGL493100_delivered_orders_obligations_transferred_unpaid": lambda col: sf.sum(col).alias(col),
+            "USSGL483200_undeliv_orders_oblig_transferred_prepaid_advanced": lambda col: sf.sum(col).alias(col),
+            "last_modified_date": lambda col: sf.max(col).alias(col),
+        }
+
+    @property
+    def object_class_program_activity_select_cols(self) -> list[Any]:
+        return [
+            col
+            for col in query_paths["object_class_program_activity"]["federal_account"].keys()
+            if not col.startswith("last_modified_date")
+        ] + ["last_modified_date"]
+
     @property
     def object_class_program_activity(self) -> DataFrame:
-        raise NotImplementedError
+        df = (
+            self._object_class_program_activity_df.filter(self.dynamic_filters)
+            .filter(
+                sf.col("submission_id").isin(
+                    get_submission_ids_for_periods(
+                        self.reporting_fiscal_year, self.reporting_fiscal_quarter, self.reporting_fiscal_period
+                    )
+                )
+            )
+            .groupBy(self.object_class_program_activity_groupby_cols)
+            .agg(*[agg_func(col) for col, agg_func in self.object_class_program_activity_agg_cols.items()])
+            # .select(self.object_class_program_activity_select_cols)
+        )
+        self.logger.info(df.columns)
+        return df.select(self.object_class_program_activity_select_cols)
```

Comment:
Some logging for debugging left in.  The logging import could also be removed.
```suggestion
        return (
            self._object_class_program_activity_df.filter(self.dynamic_filters)
            .filter(
                sf.col("submission_id").isin(
                    get_submission_ids_for_periods(
                        self.reporting_fiscal_year, self.reporting_fiscal_quarter, self.reporting_fiscal_period
                    )
                )
            )
            .groupBy(self.object_class_program_activity_groupby_cols)
            .agg(*[agg_func(col) for col, agg_func in self.object_class_program_activity_agg_cols.items()])
            .select(self.object_class_program_activity_select_cols)
        )       
```

Reply:
Good catch. I thought I removed this.